rskj-core/src/test/java/co/rsk/signing/ECDSASignerFromDiskKeyTest.java

```diff
@@ -0,0 +1,70 @@
+package co.rsk.signing;
+
+import org.ethereum.crypto.ECKey;
+import org.ethereum.crypto.SHA3Helper;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.spongycastle.util.encoders.Hex;
+
+import java.util.Arrays;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ ECDSASignerFromDiskKey.class })
+public class ECDSASignerFromDiskKeyTest {
+    ECDSASignerFromDiskKey signer;
+
+    @Before
+    public void createSigner() {
+        signer = new ECDSASignerFromDiskKey(new KeyId("an-id"), "a-file-path");
+    }
+
+    @Test
+    public void canSignWith() {
+        Assert.assertTrue(signer.canSignWith(new KeyId("an-id")));
+        Assert.assertFalse(signer.canSignWith(new KeyId("another-id")));
+    }
+
+    @Test
+    public void check() throws Exception {
+        KeyFileChecker checkerMock = mock(KeyFileChecker.class);
+        when(checkerMock.check()).thenReturn(Arrays.asList("message-1", "message-2"));
+        PowerMockito.whenNew(KeyFileChecker.class).withArguments("a-file-path").thenReturn(checkerMock);

```

Comment:
Given that you have 100% control of this class, why don't you choose a different way of testing this? I've mocked `new` before, and it's a handy tool if you absolutely need it. But here you could just inject the `KeyFileChecker` dependency, or maybe invert the relation (`KeyFileChecker` could receive a `Signer` as parameter).

Reply:
We can leave it for future refactor.